Panic on SetupIngress

Got this after `swarm leave`

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x87044e]

goroutine 150 [running]:
panic(0x1b39ba0, 0xc8200100a0)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/docker/libnetwork.(*network).addDriverWatches(0xc82090cdc0)
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/agent.go:476 +0x58e
github.com/docker/libnetwork.glob.func1(0x7f43a577f4e0, 0xc82090cdc0, 0xc82090cdc0)
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/controller.go:672 +0x264
github.com/docker/libnetwork.(*controller).NewNetwork(0xc82043e780, 0xc82074c1c0, 0x7, 0xc82074c190, 0x7, 0xc82117d8a0, 0x19, 0xc820d60d00, 0x7, 0x8, ...)
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/controller.go:662 +0x9da
github.com/docker/docker/daemon.(*Daemon).createNetwork(0xc82039c000, 0x1, 0xc82074c1c0, 0x7, 0x0, 0xc82074c1d0, 0x7, 0x0, 0xc8211aaac0, 0x1, ...)
	/go/src/github.com/docker/docker/daemon/network.go:260 +0xe26
github.com/docker/docker/daemon.(*Daemon).SetupIngress.func1(0xc82039c000, 0xc8200a1950, 0xc82074c420, 0x10, 0x10)
	/go/src/github.com/docker/docker/daemon/network.go:144 +0x1ab
created by github.com/docker/docker/daemon.(*Daemon).SetupIngress
	/go/src/github.com/docker/docker/daemon/network.go:178 +0x10b
```

@mrjana @sanimej

--- a/daemon/cluster/cluster.go
+++ b/daemon/cluster/cluster.go
@@ -719,6 +719,13 @@ func (c *Cluster) GetLocalAddress() string {
 	return c.actualLocalAddr
 }
 
+// GetListenAddress returns the listen address.
+func (c *Cluster) GetListenAddress() string {
+	c.RLock()
+	defer c.RUnlock()
+	return c.listenAddr
+}
+
 // GetAdvertiseAddress returns the remotely reachable address of this node.
 func (c *Cluster) GetAdvertiseAddress() string {
 	c.RLock()

--- a/hack/vendor.sh
+++ b/hack/vendor.sh
@@ -70,7 +70,7 @@ clone git github.com/RackSec/srslog 365bf33cd9acc21ae1c355209865f17228ca534e
 clone git github.com/imdario/mergo 0.2.1
 
 #get libnetwork packages
-clone git github.com/docker/libnetwork e69621c5fb6882627f83187ebefe7709a7211277
+clone git github.com/docker/libnetwork 6caf9022fa093e0247f9f4b572edca868c27ece3
 clone git github.com/docker/go-events 18b43f1bc85d9cdd42c05a6cd2d444c7a200a894
 clone git github.com/armon/go-radix e39d623f12e8e41c7b5529e9a9dd67a1e2261f80
 clone git github.com/armon/go-metrics eb0af217e5e9747e41dd5303755356b62d28e3ec

